docs/src/bpmn-workflows/data-flow.md

```diff
@@ -13,12 +13,106 @@ Without additional configuration, Zeebe assumes *tightly* coupled workers. That
 
 ![payload](/bpmn-workflows/payload2.png)
 
-When the worker modifies the payload, the result is merged on top-level into the workflow instance payload.
+When the worker modifies the payload, the result is merged on top-level into the workflow instance payload. In order to use *loosely* coupled job workers, the workflow can be extended by *payload mappings*.
 
-In order to use *loosely* coupled job workers, the workflow can be extended by *payload mappings* based on [JSONPath](http://goessner.net/articles/JsonPath/). Before providing the job to the worker, Zeebe applies the mappings to the payload and generates a new JSON document. Upon job completion, the same principle is applied to map the result back into the workflow instance payload:
+## Payload Mappings
 
-![payload](/bpmn-workflows/payload3.png)
+We distinguish between *input*, *output* and *merging* mappings. Input/Output mappings are used to adapt payload to the context of an activity. Merging mappings are used whenever multiple flows of execution are joined into one, for example at a merging parallel gateway.
 
-See the [Tasks section](bpmn-workflows/tasks.html) for how to define payload mappings in BPMN.
+Payload mappings are pairs of [JSONPath](http://goessner.net/articles/JsonPath/) expressions. Every mapping has a *source* and a *target* expression. The source expression describes the path in the source document from which to copy data. The target expression describes the path in the new document that the data should be copied to. When multiple mappings are defined, they are applied in the order of their appearance.
 
 **Note**: Mappings are not a tool for performance optimization. While a smaller document can save network bandwidth when publishing the job, the broker has extra effort of applying the mappings during workflow execution.
+
+### Input/Output Mappings
+
+Before starting a workflow element, Zeebe applies input mappings to the payload and generates a new JSON document. Upon element completion, output mappings are applied to map the result back into the workflow instance payload.
+
+Examples in BPMN:
+
+* Service Task: Input and output mappings can be used to adapt the workflow instance payload to the job worker.
+* Message Catch Event: Output mappings can be used to merge the message payload into the workflow instance payload.
```

Comment:
This is not implemented yet :sweat_smile: Maybe nobody sees it before the next release.